### DISCUSSION TITLE
Pre-fetch auth token in --watch mode

### DISCUSSION THREAD

**Discussion by MaxTibs** [dotansimha/graphql-code-generator#5220] [Q&A]
Hi,
First of all, I want to thank you for your amazing work! This library is awesome 😄 

I'm using graphql-code-generator to generate typing from multiple graphql servers that implement the OpenID protocol (Microservice architecture). To do so, I need to fetch an access_token from an authorization server before I use the _/grapql_ endpoints. 

How would you automatically fetch the token and pass it in the Authorization header in the _codegen.yaml_ file ?
I tried to fetch the $TOKEN in the _afterStart_ lifecycle hooks with a scripts and then use it in the Authorization header, but it is not possible since the script exporting the environment variable needs to be sourced beforehand.

Before I move on to other tools/liraries to pre-fetch the token, I want to know if this is actually possible to achieve what I ask with only this library.

Thank you :)

**Comment by dotansimha** [comment]
You can do that with the programatic usage, or use `.js` extension for your config, and fetch it :)
